source/agora/node/Runner.d

```diff
@@ -319,7 +319,7 @@ private void runDNSServer_canThrow (in RegistryConfig config, NameRegistry regis
             auto pack = udp.recv(buffer, ppeer);
             auto query = deserializeFull!Message(pack);
             registry.answerQuestions(
-                query,
+                query, peer.toAddressString,
```

Comment:
Need to consider [this](https://github.com/bosagora/agora/pull/2703#discussion_r780897531)

Reply:
I will also consider reverse proxy situation

Reply:
I've checked Nginx about reverse proxy on stream sockets (i.e. TCP and UDP) and stream configs doesn't have a `forwarded for` feature like in the HTTP configurations. Either client IP is directed or proxy IP is directed, thus we cannot do anything about it through Agora.